### ISSUE TITLE
aws_lb_target_group requires HTTP or HTTPS

### ISSUE THREAD

**Issue by jasonkuehl** [terraform-providers/terraform-provider-aws#2668]
It would appear that when you cant specify anything other then HTTP or HTTPS on a protocol in a target group. This is an issue when you try and create TCP Load Balancers.

Error: aws_lb_target_group.testexternal: "protocol" must be either "HTTP" or "HTTPS"

Terraform v0.11.1

Example Code

```
resource "aws_lb" "testexternal" {
  name                        = "testserver"

  load_balancer_type          = "network"
  internal                    = false
  subnets                     = ["${module.subnet.ELB-subnet-ids}"]
  enable_deletion_protection  = true
}

resource "aws_lb_target_group" "testexternal" {
  name     = "testexternal"
  protocol = "TCP"
  port     = 22
  vpc_id      = "${aws_vpc.bla.id}"

  health_check {
      healthy_threshold   = 10
      unhealthy_threshold = 2
      interval            = 10
      timeout             = 3
  }
}

resource "aws_lb" "testexternal" {
  name                        = "testserver"

  load_balancer_type          = "network"
  internal                    = false
  subnets                     = ["${module.subnet.ELB-subnet-ids}"]
  enable_deletion_protection  = true
}

resource "aws_lb_listener" "testexternal" {
  load_balancer_arn = "${aws_lb.testexternal.arn}"
  protocol          = "TCP"
  port              = "22"

  default_action {
    target_group_arn = "${aws_lb_target_group.testexternal.arn}"
    type             = "forward"
  }
}

resource "aws_lb_target_group_attachment" "testexternal" {
  target_group_arn = "${aws_lb_target_group.testexternal.arn}"
  target_id        = "${aws_instance.testserver-001.id}"
  port             = 22
}
```

**Comment by Techcadia** [comment]
https://github.com/terraform-providers/terraform-provider-aws/issues/1838 Looks like it was solved might be that your running an older version of the provider?

**Comment by jasonkuehl** [comment]
@Techcadia 

Thanks for the link, I just check my providers and I see I'm using 0.11.1


```
terraform {
  required_version = "0.11.1"
```

**Comment by Techcadia** [comment]
@jasonkuehl Can you check the version your are currently running of the provider.

you can run `terraform init` and it should give you the provider version its currently using.

For example
```
* provider.aws: version = "~> 1.0"
```

I have an assumption you might be running a older version like `1.0.0` and not the newest version `1.6.0`

**Comment by jasonkuehl** [comment]
@Techcadia Its showing

`provider.aws: version = “~> 1.5”`

Does my issue go away in 1.6?

**Comment by jasonkuehl** [comment]
It looks like 1.6 came out today?

https://github.com/terraform-providers/terraform-provider-aws/blob/v1.6.0/CHANGELOG.md

**Comment by jasonkuehl** [comment]
It looks like my TCP issue was fixed in 1.6 however I have a new issues in this ticket

https://github.com/terraform-providers/terraform-provider-aws/issues/2708

This ticket can be closed.

**Comment by TechIsCool** [comment]
@jasonkuehl since your the creator you can close the ticket.
